pkg/pdp/aggregator/pieceaggregator_test.go

```diff
@@ -0,0 +1,166 @@
+package aggregator_test
```

Comment:
Seeking opinions on testing strategy here. So much of this is mocked that all we are really doing is asserting the right methods are called in the right order. While this add some value, my feeling is an integration/system test would be more useful here, ideally ad a higher level in the call stack.

Reply:
@frrist there is https://github.com/storacha/testthenetwork - which runs real storage node, indexer and IPNI, mocking out only the upload service. Would be great to also have PDP storage node tested here.